File: 2026-1/sesion-de-clase-03/java/seguro-vehicular/CoberturaAsistenciaVial.java

public class CoberturaAsistenciaVial extends SeguroDecorador {

    public CoberturaAsistenciaVial(Cotizable seguro) {
        super(seguro);
    }

    @Override
    public double CalcularCosto() {
        return this.seguro.CalcularCosto() + 100.00;
    }

    @Override
    public String Descripcion() {
        return this.seguro.Descripcion() + ", asistencia vial";
    }
}

File: 2026-1/sesion-de-clase-03/dotnet/seguro-vehicular/Coberturas/CoberturaAsistenciaVial.cs
using Seguros;

namespace Coberturas;

public class CoberturaAsistenciaVial : SeguroDecorador
{
    public CoberturaAsistenciaVial(ICotizable seguro) : base(seguro)
    {
    }

    public override double CalcularCosto()
    {
        return this.seguro.CalcularCosto() + 100.00;
    }

    public override string Descripcion()
    {
        return this.seguro.Descripcion() + ", asistencia vial";
    }
}
